client/handle_http.go

```diff
@@ -2381,6 +2439,78 @@ func (te *TransferEngine) walkDirUpload(job *clientTransferJob, transfers []tran
 	return err
 }
 
+// This function performs the ls command by walking through the specified directory and printing the contents of the files
+func listHttp(ctx context.Context, remoteObjectUrl *url.URL, directorUrl string, namespace namespaces.Namespace, token string, options ...TransferOption) (fileInfos []fileInfo, err error) {
+	// Get our directory listing host
+	dirListHost, err := getDirListHost(ctx, remoteObjectUrl, namespace, directorUrl)
+	if err != nil {
+		return
+	}
+	log.Debugln("Dir list host: ", dirListHost.String())
+
+	client := createWebDavClient(dirListHost, token)
+	remotePath := remoteObjectUrl.Path
+
+	infos, err := client.ReadDir(remotePath)
+	if err != nil {
+		// Check if we got a 404:
+		if gowebdav.IsErrNotFound(err) {
+			err = errors.Wrap(err, "file not found")
```

Comment:
This might actually be a case where we don't want to wrap the error. If we get a 404, we know that it's a "not found" issue, so we can tell the user that instead of whatever ugly message pops out of the web dav client.

Reply:
The webdav client only returns this: `PROPFIND /test/fo/: 404`. I just thought things might look more user-friendly if we include just a little `object not found` or something or even rewrite the error all-together since a user might not know/care about `PROPFIND`

How about we just have this for the error (I like it better without it saying PROPFIND since it is a simple and common error): `Failure getting pelican://90ba8eb4ebfb:8444/test/thisdoesnotexist: 404: object not found`